File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-jamba-development/JMBA_InComm_iOS_SDK/InCommSDK/Models/Request/InCommSubmitOrder.swift
//
//  InCommSubmitOrder.swift
//  InCommSDK
//
//  Created by Taha Samad on 8/17/15.
//  Copyright (c) 2015 Fishbowl. All rights reserved.
//

import Foundation

public struct InCommSubmitOrder {
    
    public var purchaser: InCommOrderPurchaser
    public var recipients: [InCommOrderRecipientDetails]
    public var payment: InCommSubmitPayment?
    public var paymentWithId: InCommSubmitPaymentWithId?
    //
    public var delayActivation: Bool?
    public var description: String?
    public var discountAmount: Double?
    public var discountId: Int32?
    public var fulfillImmediately: Bool?
    public var id: String?
    public var purchaseOrderFilename: String?
    public var purchaseOrderNumber: String?
    
    public init (purchaser: InCommOrderPurchaser, recipients: [InCommOrderRecipientDetails], payment: InCommSubmitPayment?, paymentWithId: InCommSubmitPaymentWithId?) {
        self.purchaser = purchaser
        self.recipients = recipients
        self.payment   = payment
        self.paymentWithId = paymentWithId
    }
    
    public func serializeAsJSONDictionary() -> InCommJSONDictionary {
        var jsonDict = InCommJSONDictionary()
        jsonDict["Purchaser"]             = purchaser.serializeAsJSONDictionary()
        jsonDict["Recipients"]            = recipients.map { $0.serializeAsJSONDictionary() }
        
        if paymentWithId != nil{
            jsonDict["Payment"]          = paymentWithId!.serializeAsJSONDictionary()
        }
        else{
        jsonDict["Payment"]               = payment!.serializeAsJSONDictionary()
        }
        //
        
        jsonDict["DelayActivation"]       = delayActivation
        jsonDict["Description"]           = description
        jsonDict["DiscountAmount"]        = discountAmount
        if discountId != nil {
            jsonDict["DiscountId"]        = NSNumber(int: discountId!)
        }
        jsonDict["FulfillImmediately"]    = fulfillImmediately
        jsonDict["Id"]                    = id
        jsonDict["PurchaseOrderFilename"] = purchaseOrderFilename
        jsonDict["PurchaseOrderNumber"]   = purchaseOrderNumber
        return jsonDict
    }
    
}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.1/JMBA_InComm_Android_SDK/app/src/main/java/com/wearehathway/apps/incomm/Models/InCommSubmitOrder.java
package com.wearehathway.apps.incomm.Models;

import java.util.List;

/**
 * Created by Nauman Afzaal on 06/08/15.
 */
public class InCommSubmitOrder
{
    private boolean DelayActivation; //If true, PlatformServices will prevent automatic activation and require manual activation later.
    private String Description;
    private double DiscountAmount;
    private int DiscountId;
    private boolean FulfillImmediately;
    private String Id;
    private String PurchaseOrderFilename;
    private String PurchaseOrderNumber;
    private InCommOrderPurchaser Purchaser;
    private List<InCommOrderRecipientDetails> Recipients;
    private InCommSubmitPayment Payment;

    public boolean isDelayActivation()
    {
        return DelayActivation;
    }

    public void setDelayActivation(boolean delayActivation)
    {
        DelayActivation = delayActivation;
    }

    public String getDescription()
    {
        return Description;
    }

    public void setDescription(String description)
    {
        Description = description;
    }

    public double getDiscountAmount()
    {
        return DiscountAmount;
    }

    public void setDiscountAmount(double discountAmount)
    {
        DiscountAmount = discountAmount;
    }

    public int getDiscountId()
    {
        return DiscountId;
    }

    public void setDiscountId(int discountId)
    {
        DiscountId = discountId;
    }

    public boolean isFulfillImmediately()
    {
        return FulfillImmediately;
    }

    public void setFulfillImmediately(boolean fulfillImmediately)
    {
        FulfillImmediately = fulfillImmediately;
    }

    public String getId()
    {
        return Id;
    }

    public void setId(String id)
    {
        Id = id;
    }

    public String getPurchaseOrderFilename()
    {
        return PurchaseOrderFilename;
    }

    public void setPurchaseOrderFilename(String purchaseOrderFilename)
    {
        PurchaseOrderFilename = purchaseOrderFilename;
    }

    public String getPurchaseOrderNumber()
    {
        return PurchaseOrderNumber;
    }

    public void setPurchaseOrderNumber(String purchaseOrderNumber)
    {
        PurchaseOrderNumber = purchaseOrderNumber;
    }

    public InCommOrderPurchaser getPurchaser()
    {
        return Purchaser;
    }

    public void setPurchaser(InCommOrderPurchaser purchaser)
    {
        Purchaser = purchaser;
    }

    public List<InCommOrderRecipientDetails> getRecipients()
    {
        return Recipients;
    }

    public void setRecipients(List<InCommOrderRecipientDetails> recipients)
    {
        Recipients = recipients;
    }

    public InCommSubmitPayment getPayment()
    {
        return Payment;
    }

    public void setPayment(InCommSubmitPayment payment)
    {
        Payment = payment;
    }

    public  void resetPurchaser(){
        Purchaser=null;
    }
}
